src/components/structures/MatrixChat.tsx

```diff
@@ -2020,7 +2021,8 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
                     {...this.getServerProperties()}
                 />
             );
-        } else if (this.state.view === Views.LOGIN) {
+        } else if (this.state.view === Views.LOGIN
+            || (this.state.view === Views.WELCOME && shouldUseLoginForWelcome(SdkConfig.get()))) {
```

Comment:
Hmm... I guess to me, it seems _more_ confusing to present the same view under two names than it would be to properly change the view from welcome to login... Why don't you want to actually change the view for this?

Reply:
I thought it was more complicated than it actually is... Turns out it's not.